posts/2024-04-29-simplifying-nosql-database-integration-with-jakarta-nosql.adoc

```diff
@@ -0,0 +1,418 @@
+
+https://projects.eclipse.org/projects/technology.jnosql[*Eclipse JNoSQL*] is a compatible implementation of the *Jakarta NoSQL* and *Jakarta Data* specifications, a framework that streamlines the integration of Java applications with relational and NoSQL databases. Powered by the Jakarta Contexts and Dependency Injection (CDI) specification, it is compatible with Jakarta EE and Eclipse MicroProfile compatible solutions.
+
+Eclipse JNoSQL covers four NoSQL database types: *key-value*, *column*, *document* and *graph* databases.
+
+[NOTE]
+Currently, the Jakarta NoSQL doesn't define an API for Graph database types but Eclipse JNoSQL provides a Graph template to explore the specific behavior of this database type by using https://tinkerpop.apache.org/[Apache TinkerPop] as a communication layer.
+
+As of March 2024, Eclipse JNoSQL supports about https://www.jnosql.org/docs/supported_dbs.html[30 NoSQL databases].
+
+==== Why Eclipse JNoSQL?
+
+In the code samples below, note the APIs similarities and differences accross different vendors. All samples demonstrate a commonly used functionality for Document databases: a document creation and the addition of a property to that document:
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/img/logos/mongodb.png[MongoDB ,align="left" width=25%,height=25%]
+[source, java]
+----
+var document = new Document();
+document.append(name, value);
+----
+****
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/img/logos/ArangoDB.png[Arango DB,align="left" width=25%,height=25%]
+[source, java]
+----
+var baseDocument = new BaseDocument();
+baseDocument.addAttribute(name, value);
+----
+****
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/img/logos/couchbase.svg[Couchbase,align="left" width=25%,height=25%]
+[source, java]
+----
+var jsonObject = JsonObject.create();
+jsonObject.put(name, value);
+----
+****
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/img/logos/orientdb.png[Orient Project,align="left" width=25%,height=25%]
+[source, java]
+----
+var document = new ODocument("collection");
+document.field(name, value);
+----
+****
+
+With Eclipse JNoSQL, developers can use a common API to integrate with different database types, free of vendor lock-in, and with a low cognitive load during learning phases. For example, using the Document API, it's possible to switch between MongoDB and ArangoDB as needed, based on Convention Over Configuration (CoC).
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/images/home_logo.png[Eclipse JNoSQL Project,align="left" width=25%,height=25%]
+[source, java]
+----
+var entity = CommunicationEntity.of("collection");
+entity.add(name, value);
+----
+****
+
+Check out some of the Jakarta NoSQL annotations:
```

Comment:
Might be nice to have more of a link to the section below. So maybe something like...
"Check out some of the Jakarta NoSQL annotations that Eclipse JNoSQL enables:" (if this is correct?)